From 9a4b0a8ba7368fcbe847c975a6a4159f91b20f62 Mon Sep 17 00:00:00 2001
From: Tiejun Chen <tiejun.chen@windriver.com>
Date: Fri, 13 Aug 2010 17:29:56 +0800
Subject: [PATCH] ppc-mc/e500: fix incorrect debug exception entry

When support SPE we introduce a dummy handler in linux kernel in order
to create the correct SPE interrupt vector offset as the hypervisor. But
actually that should be used for debug exception is defined by guest OS.

Now fix that to be dedicated to DEBUG exception.

Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 arch/powerpc/kernel/head_wrhv.S |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/powerpc/kernel/head_wrhv.S b/arch/powerpc/kernel/head_wrhv.S
index caba46d..1233f2c 100644
--- a/arch/powerpc/kernel/head_wrhv.S
+++ b/arch/powerpc/kernel/head_wrhv.S
@@ -435,7 +435,8 @@ interrupt_base:
         START_EXCEPTION(InstructionTLBError)
         b       InstructionStorage
 
-	EXCEPTION(0x2000, DebugUnavailable, unknown_exception, EXC_XFER_EE)
+	/* Debug Interrupt */
+	DEBUG_DEBUG_EXCEPTION
 #ifdef CONFIG_SPE
 	/* SPE Unavailable */
 	START_EXCEPTION(SPEUnavailable)
@@ -464,8 +465,7 @@ interrupt_base:
 
 	CRITICAL_EXCEPTION(0x2080, CriticalDoorbell, unknown_exception)
 
-	/* Debug Interrupt */
-	DEBUG_DEBUG_EXCEPTION
+	/* Debug Crit Interrupt */
 	DEBUG_CRIT_EXCEPTION
 
 /*
-- 
1.6.5.2

